lib/Sema/ConstraintSystem.cpp

```diff
@@ -1631,6 +1690,16 @@ ConstraintSystem::getTypeOfMemberReference(
         [&](Type type) { return openType(type, replacements); });
   }
 
+  if (auto *funcDecl = dyn_cast<AbstractFunctionDecl>(value)) {
+    auto *fullFunctionType = openedType->getAs<AnyFunctionType>();
+
+    // Strip off the 'self' parameter
+    auto *functionType = fullFunctionType->getResult()->getAs<FunctionType>();
+    functionType = unwrapPropertyWrapperParameterTypes(*this, funcDecl, functionRefKind,
+                                                       functionType, locator);
+    openedType = FunctionType::get(fullFunctionType->getParams(), functionType);
```

Comment:
Is there a reason why we're not doing any FunctionType::ExtInfo propagation/preservation here? In other words, are the default ExtInfo flags the goal or do they just happen to work?

And as a general observation that long predates your pull request, it's kind of frustrating that the FunctionType constructor let's one elide (and therefore default) the ExtInfo state. This wouldn't ordinarily be a problem for most constructors but Swift is constantly unpacking and rebuilding function types and I find that the ExtInfo preservation/management to be haphazard. I've been tempted to propose something but I don't know what people think/feel about `ExtInfo`.

Reply:
Ah, I think you're right that dropping them just happens to work here. Thanks for catching - I'll fix this in a followup PR.

Agreed that the default `ExtInfo()` makes it too easy to accidentally drop existing ExtInfo when transforming a function type like this. I actually had to fix the same mistake in the body of `unwrapPropertyWrapperParameterTypes` as well

Reply:
Thanks for clarifying. A link to the followup PR would be appreciated. I just posted to the forum about making `ExtInfo` preservation/propagation more explicit:

https://forums.swift.org/t/rfc-make-extinfo-param-to-functiontype-construction-always-explicit/45182

Reply:
Just to close the loop, I merged #36234 (but GitHub thinks the merge failed, weird). Thanks for helping me better understand the intent here and now that #36234 is merged, hopefully intent will be a little more clear in the future. :-)